test/base-compiler-tests.ts

```diff
@@ -1712,3 +1712,320 @@ describe('Rust overrides', () => {
         ).toEqual(['-C', 'debuginfo=2', '-o', 'output.txt', '--crate-type', 'bin', '-Clinker=/usr/aarch64/bin/gcc']);
     });
 });
+
+// ======================================================================
+// Tests for methods that don't require complex mocking
+// ======================================================================
+describe('BaseCompiler additional method tests', () => {
+    let ce: CompilationEnvironment;
+    let compiler: BaseCompiler;
+
+    beforeEach(() => {
+        ce = makeCompilationEnvironment({languages});
+
+        const compilationInfo = makeFakeCompilerInfo({
+            exe: '/fake/compiler/path',
+            remote: undefined,
+            lang: 'c++',
+            options: '-Wall -O2',
+            version: 'clang version 13.0.0',
+            instructionSet: 'amd64',
+            supportsMarch: true,
+            supportsTarget: true,
+            supportsTargetIs: true,
+            supportsHyphenTarget: false,
+        });
+
+        compiler = new BaseCompiler(compilationInfo, ce);
+    });
+
+    describe('couldSupportASTDump', () => {
+        it('should identify clang 3.3+ as supporting AST dump', () => {
+            expect(compiler.couldSupportASTDump('clang version 3.0.0')).toBe(false);
+            expect(compiler.couldSupportASTDump('clang version 3.2.0')).toBe(false);
+            expect(compiler.couldSupportASTDump('clang version 3.3.0')).toBe(true);
+            expect(compiler.couldSupportASTDump('clang version 3.8.0')).toBe(true);
+            expect(compiler.couldSupportASTDump('clang version 13.0.0')).toBe(true);
+            expect(compiler.couldSupportASTDump('gcc version 10.2.0')).toBe(false);
+            expect(compiler.couldSupportASTDump('Some random text')).toBe(false);
+        });
+    });
+
+    describe('isCfgCompiler', () => {
```

Comment:
I think this test makes it clear we need to refactor `isCfgCompiler()` into something like `static isCfgCompilerBasedOnVersionTextAndArch(version: string, instructionSet: string)` so it doesn't actually need a basecompiler object and makes testing it much simpler